flyteplugins/go/tasks/plugins/k8s/ray/ray.go

```diff
@@ -219,7 +219,40 @@ func buildHeadPodTemplate(container *v1.Container, podSpec *v1.PodSpec, objectMe
 
 	headPodSpec := podSpec.DeepCopy()
 
-	headPodSpec.Containers = []v1.Container{*primaryContainer}
+	// Ray logs integration
+	foundTmpRayVolMount := false
+	for _, vm := range primaryContainer.VolumeMounts {
+		if vm.MountPath == "/tmp/ray" {
+			foundTmpRayVolMount = true
+			break
+		}
+	}
+	if !foundTmpRayVolMount {
+		vol := v1.Volume{
+			Name: "system-ray-state",
+			VolumeSource: v1.VolumeSource{
+				EmptyDir: &v1.EmptyDirVolumeSource{},
+			},
+		}
+		headPodSpec.Volumes = append(headPodSpec.Volumes, vol)
+	}
+
+	writeableVolMount := v1.VolumeMount{
+		Name:      "system-ray-state",
+		MountPath: "/tmp/ray",
+	}
+	primaryContainer.VolumeMounts = append(primaryContainer.VolumeMounts, writeableVolMount)
```

Comment:
This block should be included in the `if` block, right? otherwise it'll double add that volumemount if one exists...

Reply:
Yea this is wrong. I think we want to use info from an existing volume / volume mount where possible. I'll clean up.